app/scripts/lib/persistData.js

```diff
@@ -0,0 +1,27 @@
+import log from 'loglevel';
+import { captureException } from '@sentry/browser';
+
+let dataPersistenceFailing = false;
```

Comment:
before we deduplicated this logic, the two places had separate 'dataPersistenceFailing' variables. Is having them share one going to cause any issues?

Reply:
It's just used for logging. It wouldn't seem to cause any issues even for that as far as I can tell. It will update the variable on each call, and it won't impact the functionality of the next call.

Reply:
It's meant to represent cases where `storage.local` fails. Usually happens when the user's disk is full. There is an issue in the backlog somewhere for adding an error to the UI when this happens, but that work hasn't been done yet.

We're working with the same storage API here, so there isn't much sense in having two variables tracking this. Either storage is full or it isn't.